src/prommis/nanofiltration/diafiltration_two_salt.py

```diff
@@ -0,0 +1,1097 @@
+            return self.permeate_flow_volume[0] == (1e-8 * units.m**3 / units.h)
+
+        self.initial_permeate_flow_volume = Constraint(
+            rule=_initial_permeate_flow_volume
+        )
+
+        def _initial_membrane_interface_lithium(self):
+            return self.membrane_conc_mass_lithium[0, 0] == (
+                1e-10 * units.kg / units.m**3
+            )
+
+        self.initial_membrane_interface_lithium = Constraint(
+            rule=_initial_membrane_interface_lithium
+        )
+
+        def _initial_membrane_interface_cobalt(self):
+            return self.membrane_conc_mass_cobalt[0, 0] == (
+                1e-10 * units.kg / units.m**3
+            )
+
+        self.initial_membrane_interface_cobalt = Constraint(
+            rule=_initial_membrane_interface_cobalt
+        )
+
+        def _initial_membrane_interface_chlorine(self):
+            return self.membrane_conc_mass_chlorine[0, 0] == (
+                1e-10 * units.kg / units.m**3
+            )
+
+        self.initial_membrane_interface_chlorine = Constraint(
+            rule=_initial_membrane_interface_chlorine
+        )
+
+        def _initial_permeate_conc_mass_lithium(self):
+            return self.permeate_conc_mass_lithium[0] == (1e-10 * units.kg / units.m**3)
+
+        self.initial_permeate_conc_mass_lithium = Constraint(
+            rule=_initial_permeate_conc_mass_lithium
+        )
+
+        def _initial_permeate_conc_mass_cobalt(self):
+            return self.permeate_conc_mass_cobalt[0] == (1e-10 * units.kg / units.m**3)
+
+        self.initial_permeate_conc_mass_cobalt = Constraint(
+            rule=_initial_permeate_conc_mass_cobalt
+        )
+
+        def _initial_d_retentate_conc_mass_lithium_dx(self):
+            return self.d_retentate_conc_mass_lithium_dx[0] == (
+                1e-15 * units.kg / units.m**3
+            )
+
+        self.initial_d_retentate_conc_mass_lithium_dx = Constraint(
+            rule=_initial_d_retentate_conc_mass_lithium_dx
+        )
+
+        def _initial_d_retentate_conc_mass_cobalt_dx(self):
+            return self.d_retentate_conc_mass_cobalt_dx[0] == (
+                1e-15 * units.kg / units.m**3
+            )
+
+        self.initial_d_retentate_conc_mass_cobalt_dx = Constraint(
+            rule=_initial_d_retentate_conc_mass_cobalt_dx
+        )
+
+        def _initial_d_retentate_flow_volume_dx(self):
+            return self.d_retentate_flow_volume_dx[0] == (1e-15 * units.m**3 / units.h)
+
+        self._initial_d_retentate_flow_volume_dx = Constraint(
+            rule=_initial_d_retentate_flow_volume_dx
+        )
+
+    def discretize_model(self):
+        discretizer = TransformationFactory("dae.finite_difference")
+        discretizer.apply_to(
+            self, wrt=self.x_bar, nfe=self.config.NFEx, scheme="FORWARD"
+        )
+        discretizer.apply_to(
+            self, wrt=self.z_bar, nfe=self.config.NFEz, scheme="FORWARD"
+        )
```

Comment:
We're flowing from x=0 to x=1 and z=0 to z=1, right? Why are we using a forward difference instead of a backward difference?